packages/lesswrong/components/users/UsersProfile.tsx

```diff
@@ -237,6 +243,8 @@ const UsersProfileFn = ({terms, slug, classes}: {
     // maintain backward compatibility with bookmarks
     const currentSorting = (query.sortedBy || query.view ||  "new") as PostSortingMode
     const currentFilter = query.filter ||  "all"
+    const commentQueryName = "commentsSortBy"
+    const currentCommentSortBy = (query[commentQueryName]) as CommentSortingMode
```

Comment:
(I'm sure we don't handle this very well elsewhere in the codebase, but technically this is unsafe - a user can always manually type in `?commentsSortBy=foobar`.  In this case `foobar` will get passed into the view as a term, and we don't have a corresponding sort mode, so it'll probably barf when trying to sort by a database field that doesn't exist.)

Reply:
I'm not sure what the impact of being able to inject an arbitrary string as a sort criteria is (in particular I don't know whether it's SQL injection; it might be). But it does at least let you sort by fields that you don't have read access to, such as `userIP`, so this is definitely at least a little bit of a vuln.

Reply:
Presumably if we check "is this one of the designated strings" before actually submitting the query, it should be fine? I think we've done similar things elsewhere.

Reply:
does this style of type guard function seem right?

function isCommentSortingMode(value: string): value is CommentSortingMode {
  return ['new', 'old', 'top', 'controversial'].includes(value);
}

Reply:
Nvm, check the new commit called 'type check' to see how I've handled it.